### ISSUE TITLE
Cannot deploy to Amazon AWS, fatal error at parseLinks

### ISSUE THREAD

**Issue by tehmaestro** [francoispluchino/composer-asset-plugin#151]
Hi.
  When I try to deploy to Amazon, I get an error for the following command:
`export COMPOSER_HOME=/root && /usr/bin/composer.phar global require fxp/composer-asset-plugin:1.0.0-beta4`

I tried changing this to:
`export COMPOSER_HOME=/root && /usr/bin/composer.phar global require fxp/composer-asset-plugin:1.0.3` but same result.

After going `sudo` on the machine and running the exact same command, I get:
```
   PHP Fatal error:  Call to undefined method Composer\Package\Version\VersionParser::parseLinks() in /root/vendor/fxp/composer-asset-plugin/Repository/VcsPackageFilter.php on line 270

   Fatal error: Call to undefined method Composer\Package\Version\VersionParser::parseLinks() in /root/vendor/fxp/composer-asset-plugin/Repository/VcsPackageFilter.php on line 270
```

I guess this is related to the fact that I was using an old version of `composer-asset-plugin` right? Ok, then I tried to update the plugin doing: `composer global update` but it does not work, same exception.

 I also tried `/usr/bin/composer.phar global require fxp/composer-asset-plugin:1.0.3` directly as sudo, but I still get the same error. It seems I'm stuck, as I cannot update the plugin. Any ideas?

Thanks.

**Comment by francoispluchino** [comment]
Do you have tested to update Composer?
```
export COMPOSER_HOME=/root && /usr/bin/composer.phar self-update && /usr/bin/composer.phar global require fxp/composer-asset-plugin:1.0.3
```

**Comment by tehmaestro** [comment]
Yes, I tried it. Same thing. Actually, my first deployment command is always `self-update`

**Comment by francoispluchino** [comment]
Use the last stable version `1.0.3` or the dev version `1.1.0@dev` and not the `1.0.0-beta4` version (see #133).

**Comment by tehmaestro** [comment]
Yeah, I tried, but it keeps giving me the error on the current version. Should I try to remove it first?

**Comment by francoispluchino** [comment]
I also use AWS Elastic Beanstalk, and I didn't have this problem. But my config file is different.

**.ebextensions/deploy.config:**
```yaml
commands:
  01getSudo:
    command: echo Defaults:root \!requiretty >> /etc/sudoers
  02selfUpdateComposer:
    command: export COMPOSER_HOME=/root && composer.phar self-update -n
  03installComposerAssetPlugin:
    command: export COMPOSER_HOME=/root && composer.phar global require --prefer-source fxp/composer-asset-plugin:@1.0.* -n
  //...

container_commands:
  //...
  06installComposerDependencies:
    command: export COMPOSER_HOME=/root && composer.phar run-script post-install-cmd

option_settings:
  //...
  - namespace: aws:elasticbeanstalk:container:php:phpini
    option_name: composer_options
    value: --no-dev --no-scripts --optimize-autoloader
```

**Comment by tehmaestro** [comment]
Thanks for the script. Unfortunately, nothing worked.
  In the end, I fixed it by going in and `sudoedit` -ing the files modified in the pull request you mentioned (and some others) by hand. This allowed me to get rid of the fatal error, and I was able to update to `1.0.3` .
  
 Then I put `1.0.3` in my deployment script, and there you go, deployment works again. Thanks for answering. Maybe will find this info useful.
